lib/node_modules/@stdlib/plot/table/unicode/lib/props/row-separator/set.js

```diff
@@ -0,0 +1,59 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' );
+var format = require( '@stdlib/string/format' );
+
+
+// VARIABLES //
+
+var CHARACTER_LENGTH = 1;
```

Comment:
This is incorrect. Users should be able to provide emojis, etc, which may be comprised of multiple code points. Instead, you need to check for the number of grapheme clusters.

Reply:
Also, why must a row separator only be one character? Couldn't I also want a pattern? E.g., `-+-+-`, or something similar?

Reply:
> Also, why must a row separator only be one character? Couldn't I also want a pattern? E.g., `-+-+-`, or something similar?

I wanted to generalize the arguments. We have column separators and borders as well. And as they are vertical, having them span multiple characters makes things more complex. For instance, how do we place the corners and joints if we have a 3-character long vertical border? Although it does make sense for horizontal lines, but I figured it was making the API design inconsistent and messy. For instance, `borders` takes in a shorthand top-right-bottom-left. We would have to only allow the top and bottom properties to be able to span multiple characters (or grapheme clusters to be accurate).

Reply:
Single grapheme clusters are fine to use across the board for now. And understood regarding the difficulty in supporting multiple visual characters for vertical borders/separators. While horizontal support for multiple grapheme clusters would be straightforward to add, we can wait until a user requests such a feature.

Reply:
Wait, now that I think again, maybe I was seeing it the wrong way. We can have multiple characters for a vertical line. Say if it's `-*#`. We just print:

```
-
*
#
```

This way, we never have a "width/thickness" of a line to be more than a single grapheme cluster so placing corners and joints becomes straightforward. We would have to reserve joints and corners to be a single grapheme cluster though for obvious reasons

Reply:
That makes sense. Thanks for circling back!

Reply:
Also now that we are allowing multiple character strings, I was wondering if we could change the shorthand properties (`'a b c d'`) into an array (`['a', 'b', 'c', 'd']`). This would allow the user to also have spaces as part of their line characters?

Reply:
That also makes sense. Same thing: either a string or an array of strings.

Reply:
...meaning I shouldn't have to do `['a']`; I should be able to both `'a'` and `['a']` and have them both the same thing.

Reply:
Sorry. This is for the shorthand properties. Yeah, just supporting an array of strings seems reasonable.

Reply:
A very small question: if the top border is `123` and the top joint is `-`, 

Do we render:

```
1231-3123
```

or

```
1231-2312
```

Basically, do we want to skip the border character or postpone it when encountering a joint in case of multi-character lines?

Reply:
There's also a third option:

```
1231-1231
```

Maybe make this configurable? I don't think there is one "right" way to do it.

```
{
    ...,
    "rowSeparatorMode": "interpolate" | "resume" | "repeat",
    ...,
}
```

Option name and "modes" are suggestive; if there are better alternatives, that would be fine. There may be something equivalent in the CSS world.

Reply:
As this will be a thing with every line, `lineSeparatorMode` makes more sense?

Reply:
Works for me.

Reply:
For more configurability, can we also allow it to be an array of 2 elements or a string? So they can specify separate modes for vertical and horizontal lines..?

Reply:
Maybe two separate options? `horizontalSeparatorMode` and `verticalSeparatorMode`?